### PR TITLE
Documentation for Form hooks

### PR DESCRIPTION
**Pull request by alanhartless**
This PR ports over https://developer.mautic.org/#js-form-processing-hooks

### CODE REVIEW

File: docs/form_hooks/response_hooks.rst [hunk 1]
```diff
@@ -1,3 +1,72 @@
 Response hooks
 ==========================================================
 
+.. js:method:: onResponse()
+
+    Called prior to the response form the Form submission is processed that handles downloading files, displaying a success message, and/or redirecting to another URL.
```
Comment (RCheesley) [inline docs/form_hooks/response_hooks.rst:6]:
```suggestion
    Called prior to the response form, the Form submission is processed, which handles downloading files, displaying a success message, and/or redirecting to another URL.
```

Comment (RCheesley) [reply docs/form_hooks/response_hooks.rst:6]: I think this is correct grammar, wasn't 100% sure on the context for the sentence.

File: docs/form_hooks/validation_hooks.rst [hunk 2]
```diff
@@ -1,3 +1,101 @@
 Validation hooks
 ==========================================================
 
+.. js:method:: onValidate()
+
+    Called prior to default Form validation. This is useful if you want to execute custom code prior to the Form being submitted or if you want to do your own validation.
```
Comment (RCheesley) [inline docs/form_hooks/validation_hooks.rst:6]:
```suggestion
    Called prior to default Form validation. This is useful if you want to execute custom code prior to the Form being submitted, or if you want to do your own validation.
```

**Review by RCheesley — CHANGES_REQUESTED** [review]
Just a couple of small grammatical things :)

File: docs/form_hooks/response_hooks.rst (hunk 1 above)
Comment (alanhartless) [reply docs/form_hooks/response_hooks.rst:6]: That `form` is supposed to be `from` but the whole sentence is janky. I’ll fix it.

**Review by RCheesley — APPROVED** [review]
Looks good to me! :)

**Pull request merged**